app/controllers/sys_controller.rb

```diff
@@ -76,19 +76,39 @@ def fetch_changesets
   end
 
   def repo_auth
-    @project = Project.find_by_identifier(params[:repository])
-
-    if (%w(GET PROPFIND REPORT OPTIONS).include?(params[:method]) &&
-        @authenticated_user.allowed_to?(:browse_repository, @project)) ||
-       @authenticated_user.allowed_to?(:commit_access, @project)
+    project = Project.find_by_identifier(params[:repository])
+    if authorized?(project, @authenticated_user)
       render text: 'Access granted'
-      return
+    else
+      render text: 'Not allowed', status: 403 # default to deny
+    end
+  end
+
+  private
+
+  def authorized?(project, user)
+    if git_auth?
+      authorized_with_git?(project, user, params[:uri], params[:location])
+    else
+      authorized_with_subversion?(project, user, params[:method])
     end
+  end
 
-    render text: 'Not allowed', status: 403 # default to deny
+  def git_auth?
+    params[:git_smart_http] == '1'
   end
 
-  protected
+  def authorized_with_git?(project, user, uri, location)
+    read_only = !%r{^#{location}/*[^/]+/+(info/refs\?service=)?git\-receive\-pack$}o.match(uri)
+    return (read_only && user.allowed_to?(:browse_repository, project)) ||
+     user.allowed_to?(:commit_access, project)
+  end
+
+  def authorized_with_subversion?(project, user, method)
```

Comment:
Unused method argument - `method`. If it's necessary, use `_` or `_method` as an argument name to indicate that it won't be used.